extensions/kafka/src/main/java/io/deephaven/kafka/ingest/GenericRecordArrayFieldCopier.java

```diff
@@ -0,0 +1,154 @@
+                        ++i;
+                    }
+                    return out;
+                };
+            }
+            // There is no "SHORT" in Avro.
+
+            if (componentType.equals(int.class)) {
+                return (GenericArray<?> ga) -> {
+                    final int gaSize = ga.size();
+                    if (gaSize == 0) {
+                        return EMPTY_INT_ARRAY;
+                    }
+                    final int[] out = new int[gaSize];
+                    int i = 0;
+                    for (Object o : ga) {
+                        out[i] = (o == null) ? NULL_INT : (int) o;
+                        ++i;
+                    }
+                    return out;
+                };
+            }
+            if (componentType.equals(long.class)) {
+                return (GenericArray<?> ga) -> {
+                    final int gaSize = ga.size();
+                    if (gaSize == 0) {
+                        return EMPTY_LONG_ARRAY;
+                    }
+                    final long[] out = new long[gaSize];
+                    int i = 0;
+                    for (Object o : ga) {
+                        out[i] = (o == null) ? NULL_LONG : (long) o;
+                        ++i;
+                    }
+                    return out;
+                };
+            }
+            if (componentType.equals(float.class)) {
+                return (GenericArray<?> ga) -> {
+                    final int gaSize = ga.size();
+                    if (gaSize == 0) {
+                        return EMPTY_FLOAT_ARRAY;
+                    }
+                    final float[] out = new float[gaSize];
+                    int i = 0;
+                    for (Object o : ga) {
+                        out[i] = (o == null) ? NULL_FLOAT : (float) o;
+                        ++i;
+                    }
+                    return out;
+                };
+            }
+            if (componentType.equals(double.class)) {
+                return (GenericArray<?> ga) -> {
+                    final int gaSize = ga.size();
+                    if (gaSize == 0) {
+                        return EMPTY_DOUBLE_ARRAY;
+                    }
+                    final double[] out = new double[gaSize];
+                    int i = 0;
+                    for (Object o : ga) {
+                        out[i] = (o == null) ? NULL_DOUBLE : (double) o;
+                        ++i;
+                    }
+                    return out;
+                };
+            }
+            if (componentType.equals(boolean.class)) {
+                return (GenericArray<?> ga) -> convertObjectArray(ga, EMPTY_BOOLEANBOXED_ARRAY, boolean.class);
+            }
+            if (componentType.equals(String.class)) {
+                return (GenericArray<?> ga) -> convertObjectArray(ga, EMPTY_STRING_ARRAY, String.class);
+            }
+            throw new IllegalStateException("Unsupported component type " + componentType);
```

Comment:
Do we support datetime arrays? Generic objects?

Reply:
No support for DateTime arrays as such; they will show up as long[].
My vote is for not adding more complexity now, it should be a relative easy add but needs testing etc (I am willing to make some bets on testing, like, if long[] works then int[] works, etc, I am not willing to bet on DateTime arrays, that will need testing, which needs modification of the producer kafka scripts... etc